external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaSourceProviderSuite.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.kafka010
+
+import java.util.Locale
+
+import scala.collection.JavaConverters._
+
+import org.scalatest.PrivateMethodTester
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.sources.v2.reader.Scan
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+class KafkaSourceProviderSuite extends SparkFunSuite with PrivateMethodTester {
+  private val pollTimeoutMsMethod = PrivateMethod[Long]('pollTimeoutMs)
+
+  test("SPARK-28142 - continuous mode - options should be handled as case-insensitive") {
+    // we're trying to read the value of "pollTimeout" to see whether option is handled correctly
+
+    // upper-case
+    val expectedValue = 1000
+    val mapWithUppercase = buildCaseInsensitiveStringMap(
+      KafkaSourceProvider.CONSUMER_POLL_TIMEOUT.toUpperCase(Locale.ROOT) -> expectedValue.toString)
+    assert(expectedValue === getFieldFromContinuousStream(mapWithUppercase, pollTimeoutMsMethod))
+
+    // lower-case
+    val mapWithLowercase = buildCaseInsensitiveStringMap(
+      KafkaSourceProvider.CONSUMER_POLL_TIMEOUT.toLowerCase(Locale.ROOT) -> expectedValue.toString)
+    assert(expectedValue === getFieldFromContinuousStream(mapWithLowercase, pollTimeoutMsMethod))
+  }
+
+  private def buildCaseInsensitiveStringMap(
+      options: (String, String)*): CaseInsensitiveStringMap = {
+    val requiredOptions = Map("kafka.bootstrap.servers" -> "dummy", "subscribe" -> "dummy")
+    new CaseInsensitiveStringMap((options.toMap ++ requiredOptions).asJava)
+  }
+
+  private def getKafkaDataSourceScan(options: CaseInsensitiveStringMap): Scan = {
```

Comment:
Shall we remove this function since this is used once?

Reply:
#24967 is planned to reuse this and some more methods being introduced here, as similar tests are required to be added there. @gaborgsomogyi and I discussed to bring methods in this PR. I'm happy to add UT #24967 is about to bring after merging this patch if we want to see how these methods will be reused.

Reply:
I'm not sure if #24967 is merged or not. But, I got it.